src/streamlit_app.py

```diff
@@ -483,71 +493,100 @@ def create_grouped_bar_chart(
     # Bug in streamlit with dots in column names - update column names
     updated_columns = {col: col.replace(".", "_") for col in df.columns}
     df = df.rename(columns=updated_columns)
     numeric_columns = list(df.select_dtypes(include=["number"]).columns)
+    
     if len(numeric_columns) == 0:
-        return
+        st.markdown("Chart currently not available for non-numeric data")
+        return False
+    
     # Bug in snowflake connector - does not set date types correctly in pandas, so manually detecting
     date_columns = get_possible_date_columns(df)
     str_columns = df.select_dtypes(include=["object"]).columns
     str_columns = [col for col in str_columns if col not in date_columns]
     df_to_show = df[numeric_columns + str_columns]
+    
     for col in date_columns[:1]:
         df_to_show[col] = pd.to_datetime(df[col], errors="coerce")
+
+    # date        
     if len(date_columns) > 0:
-        df_to_show = df_to_show.rename(columns={date_columns[0]: "index"}).set_index(
-            "index",
-        )
+        df_to_show = df_to_show.rename(columns={date_columns[0]: "index"}).set_index("index")
+
         if len(str_columns) == 0:
             st.line_chart(df_to_show)
         elif len(str_columns) == 1 and len(numeric_columns) == 1:
-            st.line_chart(
-                df_to_show,
-                y=numeric_columns[0],
-                color=str_columns[0],
-            )
+            st.line_chart(df_to_show, y=numeric_columns[0], color=str_columns[0])
+
+    # text
     elif len(str_columns) >= 1:
-        create_grouped_bar_chart(df_to_show, str_columns, numeric_columns)
+
+        if len(df_to_show) > 50:
+            metric_column = numeric_columns[0]  # First numeric column
+            st.markdown(f"**Showing top 50 values sorted by {metric_column}**")
+            df_to_show = df_to_show.sort_values(by=metric_column, ascending=False, ).head(50)
+
+        if (len(str_columns) == 1):
+            chart = alt.Chart(df_to_show).mark_bar().encode(
+                y=alt.X(str_columns[0], sort=alt.EncodingSortField(field=metric_column, order="descending")),
+                x=alt.Y(numeric_columns[0]),
+                color=alt.Color(str_columns[0], legend=None)  # Hide legend
+            )
+
+            st.altair_chart(chart, use_container_width=True)
+            
+        else:
+            return create_grouped_bar_chart(df_to_show, str_columns, numeric_columns)
+
+    # multiple numeric
     elif len(numeric_columns) > 1:
-        create_grouped_bar_chart(
-            df_to_show,
-            [numeric_columns[0]],
-            numeric_columns[1:],
-        )
+        return create_grouped_bar_chart(df_to_show, [numeric_columns[0]], numeric_columns[1:])
+
+    # single numeric
     elif len(numeric_columns) == 1:
         if len(df_to_show) == 1:
-            value = f"{df[numeric_columns[0]].iloc[0]:,}"
+            value = "{:,}".format(df[numeric_columns[0]].iloc[0])
```

Comment:
This was a static code analysis fix - did it introduce a bug? If not suggest to revert..